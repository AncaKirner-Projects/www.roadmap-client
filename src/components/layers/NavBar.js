import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import IconButton from '@material-ui/core/IconButton';

class NavBar extends Component {
  state = {
    open: false
  };

  render() {
    const { classes } = this.props;
    const { open } = this.state;
    return (
      <AppBar
        position="fixed"
        className={classNames(classes.appBar, {
          [classes.appBarShift]: open
        })}
      >
        <Toolbar disableGutters={!open}>
          <IconButton
            color="inherit"
            aria-label="Open drawer"
            className={classNames(classes.menuButton, {
              [classes.hide]: open
            })}
          />
        </Toolbar>
      </AppBar>
    );
  }
}

NavBar.propTypes = {
  classes: PropTypes.object.isRequired
};

export default (NavBar);
