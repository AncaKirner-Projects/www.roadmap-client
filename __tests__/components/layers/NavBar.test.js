import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NavBar from '../../../src/components/layers/NavBar';

describe('<NavBar />', () => {
  const props = {
    classes: {
      appBar: {},
      appBarShift: {},
      menuButton: {},
      hide: {}
    }
  };
  let wrapper;

  beforeAll(() => {
    wrapper = shallow(<NavBar classes={props.classes} />);
  });

  it('renders 1 component', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('renders 1 <AppBar/> component', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
