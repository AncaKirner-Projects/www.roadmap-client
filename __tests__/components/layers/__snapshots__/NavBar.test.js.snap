// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> renders 1 <AppBar/> component 1`] = `
<WithStyles(AppBar)
  className=""
  position="fixed"
>
  <WithStyles(Toolbar)
    disableGutters={true}
  >
    <WithStyles(IconButton)
      aria-label="Open drawer"
      className=""
      color="inherit"
    />
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;
